s3/src/main/java/com/instaclustr/kafka/connect/s3/TransferManagerProvider.java

```diff
@@ -30,9 +36,34 @@ public static AmazonS3ClientBuilder getS3ClientBuilderWithRegionAndCredentials(f
         String accessKey = getFromConfigOrEnvironment(config, AwsStorageConnectorCommonConfig.AWS_ACCESS_KEY_ID);
         String secret = getFromConfigOrEnvironment(config, AwsStorageConnectorCommonConfig.AWS_SECRET_KEY);
         String region = getFromConfigOrEnvironment(config, AwsStorageConnectorCommonConfig.AWS_REGION);
+        String roleArn = getFromConfigOrEnvironment(config, AwsStorageConnectorCommonConfig.AWS_IAM_ROLE_ARN);
 
-        AmazonS3ClientBuilder clientBuilder = AmazonS3ClientBuilder.standard()
-                .withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials(accessKey, secret)));
+        AmazonS3ClientBuilder clientBuilder;
+
+        if (roleArn == null || StringUtils.isBlank(roleArn)) {
+            // authenticate with access key/secret
+            AWSStaticCredentialsProvider awsStaticCredentialsProvider = new AWSStaticCredentialsProvider(new BasicAWSCredentials(accessKey, secret));
+            clientBuilder = AmazonS3ClientBuilder.standard()
+                    .withCredentials(awsStaticCredentialsProvider);
+        } else {
+            // authenticate with access key/secret, then assume role
+            AWSSecurityTokenService awsSecurityTokenService = AWSSecurityTokenServiceClientBuilder.standard()
+                    .withCredentials(new AWSStaticCredentialsProvider(new BasicAWSCredentials(accessKey, secret)))
+                    .build();
+
+            STSAssumeRoleSessionCredentialsProvider.Builder assumeRoleBuilder =
+                    new STSAssumeRoleSessionCredentialsProvider.Builder(
+                            roleArn,
+                            UUID.randomUUID().toString().substring(0, 32));
+
+            STSAssumeRoleSessionCredentialsProvider credentialsProvider = assumeRoleBuilder
+                    .withStsClient(awsSecurityTokenService)
+                    .withRoleSessionDurationSeconds((int) TimeUnit.HOURS.toSeconds(1))
```

Comment:
what's the role of duration here? I imagine we want to assume the role as long as the connector is in operation?

Reply:
yeah, Assume Role Sessions require a duration. if not provided, it defaults to 15 minutes. I am unsure what happens once the session expires, and if the connector is still active. Trying to read up on that

Reply:
thanks for drawing my attention to that piece of code @h3nd24 . 
The minimum and default assume role session duration is 15 minutes, and max is 12 hours or whatever the AWS admin has set. However, as per my testing, the S3 client seems to auto renew the session and we dont need to do anything programatically. Hence have removed the code altogether and let fallback on default, as it doesnt matter.

Reply:
Yeah, we might need an auto renewal if it's not already handled by the aws clients